File: e2e/playwright-typescript/tests/delete-task/delete-task.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.TASK_MANAGEMENT);
  await allure.feature('Delete task');
  await allure.tms('99');
});

test.describe('Delete task', () => {
  let context: TaskContext;

  test.beforeEach(async ({ support }) => {
    context = new TaskContext();
    const response = context.createTaskResponse();

    await support.mock.api.getTasks([response]);
    await support.mock.api.deleteTask(response.id);
  });

  test('should delete task', async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();

    // when
    await step.tasks.deleteTask(context.title);

    // then
    await validate.tasks.hasNoTask(context.title);
  });
});

File: e2e/capybara-ruby/spec/delete_task/delete_task_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'Delete task', type: :system,
                              epic: AllureEpic::TASK_MANAGEMENT,
                              feature: 'Delete task',
                              tms: '99' do
  let(:context) { TaskContext.new }

  before do
    response = context.create_task_response
    support.mock.api
         .get_tasks([response])
         .delete_task(response[:id])
  end

  it 'should delete task' do
    # given
    step.navigation.open_main_page

    # when
    step.tasks.delete_task(context.title)

    # then
    validate.tasks.has_no_task(context.title)
  end
end
